utils/timezone.jsx

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import UserStore from 'stores/user_store.jsx';
+import * as UserActions from 'actions/user_actions.jsx';
+
+const dateTimeFormat = new Intl.DateTimeFormat();
+
+export function getSupportedTimezones() {
+    if (global.mm_config && global.mm_config.SupportedTimezones) {
+        return global.mm_config.SupportedTimezones.split(',');
+    }
+    return [];
+}
+
+export function getBrowserTimezone() {
+    return dateTimeFormat.resolvedOptions().timeZone;
+}
+
+export function autoUpdateTimezone(userTimezone) {
```

Comment:
Couple suggestions/comments here:
1. Is any of this logic shared by the mobile app? Wonder if it would make sense to put it into mattermost-redux
2. Since an action is getting called here, I'd prefer if we just turned this function into an action itself

Reply:
Doable, I currently have a PR for timezone support so adding selectors/actions for this won't be hard.

I agree this logic is shared with both webapp and mobile